File: Tipos Datos.cpp
/*
* 1. Programa para valorar el espacio de memoria que utiliza una variable.

* 2. Conversion implicita y explicita de datos
*/

#include <iostream>

using namespace std;
int main() {

	int entero = 56;
	double doble = 10;
	float flotante = 10.0f;
	char caracter = 'A';//Tipo de entero
	char c2 = 64;
	bool boleano = true;

		cout << "Tamanio en bytes de los tipos de datos primitivos\n";
		cout << "\nEntero (int): " << entero << " (Tamanio: " << sizeof(entero) << "bytes)";
		cout << "\nDoble (double): " << doble << " (Tamanio: " << sizeof(doble) << "bytes)";
		cout << "\nFlotante (float): " << flotante << " (Tamanio: " << sizeof(flotante) << "bytes)";
		cout << "\nCaracter (char): " << caracter << " (Tamanio: " << sizeof(caracter) << "bytes)";
		cout << "\nCaracter 2 (char): " << c2 << " (Tamanio: " << sizeof(c2) << "bytes)";
		cout << "\nBooleano (bool): " << boleano << " (Tamanio: " << sizeof(boleano) << "bytes)\n";

		string cadena = "Hola chicos";
		cout << "Tamanio en bytes de los tipos de datos ESTRUCTURADO";
		cout << "\nCadena (string): " << cadena << " (Tamanio: " << sizeof(cadena) << "bytes)\n";

		cout << "\nCONVERSION DE TIPOS\n";
		//Conversion implicita: El compilador de forma automatica realiza el cambio
		flotante = entero;
		cout << "Conversion de int --> float:" << flotante; cout << endl;

		//Conversion explicita:Se obliga al compilador a realizar una conversion
		int ascii = int(caracter);
		cout << "Conversion de char --> float:" << ascii; cout << endl;


		return 0;
}

File: Tipos Datos.py
import sys

entero = 56
doble = 10
flotante = 10.0
caracter = 'A'
c2 = 64
boleano = True

print("Tamanio en bytes de los tipos de datos primitivos\n")
print("Entero (int): ",entero," (Tamanio: ", sys.getsizeof(entero), "bytes)")
print("Doble (no hay double, es float): ",doble, " (Tamanio: ", sys.getsizeof(flotante), "bytes)")
print("Flotante (float): ",flotante," (Tamanio: ", sys.getsizeof(flotante), "bytes)")
print("Caracter (char): ",caracter," (Tamanio: ", sys.getsizeof(caracter), "bytes)")
print("Caracter 2 (char): ",c2," (Tamanio: ", sys.getsizeof(c2), "bytes)")
print("Booleano (bool): ",boleano," (Tamanio: ", sys.getsizeof(boleano), "bytes)")

cadena = "\nHola chicos de Algoritmos y Programacion"
print("\nCadena (str): ",cadena," (Tamanio: ", sys.getsizeof(cadena)," bytes)")

print("\nCONVERSION DE TIPOS")

valor = "2026"
tiempo = int(valor) -45
print(tiempo)

a = input("Ahora al tiempo obtenido: " + str(tiempo)+ " ingresa tu edad\n")
print(a)